### PR TITLE
Update AdjointGenerator.h with maximum and minimum intrinsics

### PR DESCRIPTION
**Pull request by motabbara**

### CODE REVIEW

**Comment by wsmoses** [comment]
Can you add a forward and reverse mode ll test? (look inside repo/enzyme/test/ReverseMode/ and repo/enzyme/test/ForwardMode).

**Comment by motabbara** [comment]
Added a couple of tests @wsmoses

**Comment by motabbara** [comment]
How's that look @wsmoses ?

**Comment by motabbara** [comment]
Along similar lines, is maxnum.ll correct in forward mode?

**Comment by wsmoses** [comment]
@motabbara are you able to see the results of CI or run the tests locally?

I'm fine to merge as soon as the pass (and as a consequence test for the correct result :P )

File: enzyme/test/Enzyme/ForwardMode/maximum.ll [hunk 1]
```diff
@@ -0,0 +1,29 @@
+; RUN: %opt < %s %loadEnzyme -enzyme -enzyme-preopt=false -S | FileCheck %s
```
Comment (wsmoses) [inline enzyme/test/Enzyme/ForwardMode/maximum.ll:1]: The old pass manager line needs to be version gated to be < 16.

Separately if this intrinsic is only LLVM15+, you also need to version gate all tests to be 15+.

File: enzyme/test/Enzyme/ForwardMode/maximum.ll [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+declare float @llvm.maximum(float, float)
+
+; Function Attrs: nounwind
+declare float @__enzyme_fwddiff(float (float, float)*, ...)
+
+; CHECK: define internal {{(dso_local )?}}float @fwddiffetester(float %x, float %"x'", float %y, float %"y'")
+; CHECK-NEXT: entry:
+; CHECK-NEXT:   %0 = fcmp fast olt float %x, %y
+; CHECK-NEXT:   %1 = select {{(fast )?}}i1 %0, float %"x'", float %"y'"
+; CHECK-NEXT:   ret float %1
```
Comment (wsmoses) [inline enzyme/test/Enzyme/ForwardMode/maximum.ll:27]: This looks to be computing the derivative for min?

Since it returns (x < y) ? x' : y'

Comment (motabbara) [reply enzyme/test/Enzyme/ForwardMode/maximum.ll:27]: I found that confusing too. This test is identical to that for maxnum that is passes when I run locally. Is it that the forward mode implementation for that needs a fix?

Comment (motabbara) [reply enzyme/test/Enzyme/ForwardMode/maximum.ll:29]: Changed the implementation but have only updated a few of the tests.

**Review by tgymnich — APPROVED** [review]
Looks good. Thank you!

**Comment by motabbara** [comment]
@tgymnich don't think this will get auto-merged while CI pipeline is still in flux. Can you please manually merge?

**Pull request merged**